### PR TITLE
#2409049 Variable add action added

### PR DESCRIPTION
**Pull request by dermario**
https://www.drupal.org/node/2409049

My work is based on https://github.com/fago/rules/pull/95 by dawehner.

### CODE REVIEW

File: src/Plugin/Action/AddVariable.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ *     ),
+ *   },
+ *   provides = {
+ *     "variable_added" = @ContextDefinition("any",
+ *        label = @Translation("Added variable")
+ *      )
+ *    }
+ * )
+ */
+class AddVariable extends RulesActionBase {
```
Comment (fago) [inline src/Plugin/Action/AddVariable.php]: You should add a category to the action and prefix the class name with the category - as other classes do.

File: src/Plugin/Action/AddVariable.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+  /**
+   * Executes the plugin.
```
Comment (fago) [inline src/Plugin/Action/AddVariable.php]: inheritdoc. as others do

File: tests/src/Integration/Action/AddVariableTest.php [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+class AddVariableTest extends RulesIntegrationTestBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+  }
+
+  public function testExecute() {
```
Comment (fago) [inline tests/src/Integration/Action/AddVariableTest.php]: misses comment

**Comment by dermario** [comment]
I updated this branch according to your comments.

**Comment by fubhy** [comment]
Thanks. Merged!

**Pull request merged**